pyarr/radarr_api.py

```diff
@@ -0,0 +1,998 @@
+
+    # GET /moviefile
+    def get_movie_files_by_movie_id(self, id_):
+        """Returns the movie files belonging to a specific movie based on database ID.
+
+        Args:
+            [Required] id_ (int)
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"movieid": id_}
+        path = "/api/v3/moviefile"
+        res = self.request_get(path, params=params)
+        return res
+
+    # GET /moviefile
+    def get_movie_files(self, moviefile_ids):
+        """Returns movie file info for multiple movie files.
+
+        Args:
+            [Required] moviefile_ids (array:int)
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"moviefileids": moviefile_ids}
+        path = "/api/v3/moviefile"
+        res = self.request_get(path, params=params)
+        return res
+
+    # GET /moviefile/{id}
+    def get_movie_file(self, id_):
+        """Returns movie file info.
+
+        Args:
+            [Required] id_ (int)
+        Returns:
+            JSON Response (dict)
+        """
+        path = f"/api/v3/moviefile/{id_}"
+        res = self.request_get(path)
+        return res
+
+    # DELETE /moviefile/{id}
+    def del_movie_file(self, id_):
+        """Allows for deletion of a moviefile by its database ID.
+        Args:
+            [Required] id_ (int)
+        Returns:
+            JSON Response (dict)
+        """
+        path = f"/api/v3/movie/{id_}"
+        res = self.request_del(path)
+        return res
+
+    ## HISTORY
+
+    # GET /history
+    def get_history(
+        self, page, page_size=20, sort_direction="descending", sort_key="date"
+    ):
+        """Return a json object list of items in your history
+
+        Args:
+            [Required] page (int)
+            [Optional] page_size (int) - Default: 20
+            [Optional] sort_direction (str) - Default: descending
+            [Optional] sort_key (str) - Default: date
+        Returns:
+            JSON Response (dict)
+        """
+        params = {
+            "page": page,
+            "pageSize": page_size,
+            "sortDirection": sort_direction,
+            "sortKey": sort_key,
+        }
+        path = "/api/v3/history"
+        res = self.request_get(path, params=params)
+        return res
+
+    # GET /history/movie
+    def get_movie_history(self, id_, event_type=None):
+        """Return a json object list of items in your history.
+
+        Args:
+            [Required] id_ (int) - Database ID of movie
+            [Optional] event_type (int) - History event type to retrieve
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"movieId": id_}
+        if event_type:
+            params["eventType"] = event_type
+        path = "/api/v3/history/movie"
+        res = self.request_get(path, params=params)
+        return res
+
+    ## BLACKLIST
+
+    # GET /blacklist
+    def get_blacklist(
+        self,
+        page,
+        page_size=20,
```

Comment:
Set a default to match sonarr `page=1, page_size=10`

Reply:
For now I have left page_size = 20 to follow Radarr API defaults on these endpoints. 

Let's discuss whether we should follow the the Radarr docs, or match Sonarr docs.